File: java/476.java
class Solution {
    public int findComplement(int num) {
        int ans = 0;
        int i = 0;
        while(num > 0){
            int k = 1 - (num&1);
            ans += k * (int)Math.pow(2,i++);
            num >>= 1;
        }
        return ans;
    }
}

File: go/476.go
import "math"

func findComplement(num int) int {
	ans, i := 0, 0
	for num > 0 {
		k := 1 - (num & 1)
		ans += k * int(math.Pow(2, float64(i)))
		i++
		num >>= 1
	}
	return ans
}